README.md

```diff
@@ -88,9 +87,124 @@ terraform apply -parallelism=1
+
+- `./testvars`
+  * contains the generated YAML files with the test variables used in the *_test.go and example files
+
+The below steps should be followed for developing `terraform-plugin-framework` resources and data-sources:
+
+1. Create issue and comment that you will be working on the issue.
+
+2. Fork the terraform-provider-aci repository.
+
+3. Clone the forked code to your local machine.
+
+4. Add minimum required class details to [classes.yaml](https://github.com/CiscoDevNet/terraform-provider-aci/tree/master/examples/gen/definitions/classes.yaml).
+  * `ui_location`
+  * `sub_category`
+  * `tagAnnotation` is added to children when this is allowed to be configure under the class 
+
+5. Copy new meta file(s) or replace existing with newer version of the meta file(s) to the [meta](https://github.com/CiscoDevNet/terraform-provider-aci/tree/master/examples/gen/meta) directory in the following format: `<classname>.json`. Assure that all relationship classes are also added to the `meta` directory. The `GEN_HOST` and `GEN_CLASSES` environment variables can be leveraged to retrieve the classes automatically before rendering the code.
+  * `GEN_HOST` should be set to a resolvable host, example: `173.36.219.70`. When not provided the devnet documentation will be retrieved.
+  * `GEN_CLASSES` should be set to a comma-separated string of classes, example: `fvTenant` or `fvTenant,fvBD`
+
+6. Run `go generate` in the root of the local repository where the `main.go` is located. The following files should be created or updated:
+  * `resource_<resource-name>.go` and `resource_<resource-name>_test.go`
+  * `data_source_<resource-name>.go` and `data_source_<resource-name>_test.go`
+  * `<resource-name>.md` in the [docs/resources](https://github.com/CiscoDevNet/terraform-provider-aci/tree/master/docs/resources) and [docs/data-sources](https://github.com/CiscoDevNet/terraform-provider-aci/tree/master/docs/data-sources) directories.
+  * `resource.tf` and `data-source.tf`
+  * `provider.go`
+
+6. Validate generated files generated on the following items:
+  * resource name
+    * incorrect resource names should be overwritten in [classes.yaml](https://github.com/CiscoDevNet/terraform-provider-aci/tree/master/examples/gen/definitions/classes.yaml) by providing `resource_name: "<resource-name>"`
+  * schema attributes ( class properties and children )
+    * incorrect property names should be overwritten in [properties.yaml](https://github.com/CiscoDevNet/terraform-provider-aci/tree/master/examples/gen/definitions/properties.yaml) by providing a global or class specific overwrite
+    * all relationships classes are included
+  * documentation contains the example and all information is correct
+  * Run `go generate` again until the files are generated as preferred. When you are not achieving the desired outputs, reource, data-source, example and/or documentation templates should be adjusted.
```

Comment:
It would be good to have a troubleshooting section, like this:

Troubleshooting
If you encounter an error message while generating the resources using go generate, a few steps can be followed:

Make sure that you're running the latest version of Go

Update dependencies and populate the vendor directory: If you're using Go modules, you can update your dependencies and populate your vendor directory by running the following commands in your terminal:
```
go mod tidy
go mod vendor
```
The go mod tidy command will clean up unused dependencies and add missing ones. The go mod vendor command will copy all dependencies into the vendor directory.

Disable vendor mode (if necessary): If the error still persists, consider disabling vendor mode by setting GOFLAGS="-mod=mod", and then run go get again:
```
export GOFLAGS="-mod=mod"
```
This will force Go to fetch the package directly, regardless of the vendor directory. After the package is downloaded, you can switch back to vendor mode if needed.

Reply:
ok added